staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset_test.go

```diff
@@ -239,22 +394,33 @@ func TestUniformFlows(t *testing.T) {
 	}
 	qs := qsc.Complete(fq.DispatchingConfig{ConcurrencyLimit: 4})
 
-	exerciseQueueSetUniformScenario(t, qCfg.Name, qs, []uniformClient{
-		{1001001001, 5, 10, time.Second, time.Second},
-		{2002002002, 5, 10, time.Second, time.Second},
-	}, time.Second*20, true, true, true, true, "", clk, counter)
+	uniformScenario{name: qCfg.Name,
+		qs: qs,
+		clients: []uniformClient{
+			{1001001001, 8, 20, time.Second, time.Second - 1, false},
```

Comment:
This sort of configuration gets into a cadence where every 1000000000 nanoseconds some requests finish, and 1 ns before that some more arrive.  That means that when one request finishes, the dispatcher already has all the requests it will consider.  If the arrivals were at the same fake time as the completions then we are at the mercy of the golang goroutine scheduler: some dispatching may happen before some of the request arrivals.